src/xref2/ref_tools.ml

```diff
@@ -831,6 +831,11 @@ let resolved3 (r, _, _) = resolved1 r
 
 and resolved2 (r, _) = resolved1 r
 
+let resolve_asset_reference env (r : Reference.Asset.t) : Asset.t ref_result =
+  match r with
+  | `Resolved _r -> failwith "What's going on!?"
```

Comment:
Fixed for reresolving assets in 01109c6af
The strange failwith is still there for module and signature references, since it's harder to solve and out of scope from this PR.